src/cargo/ops/cargo_remove.rs

```diff
@@ -33,7 +34,17 @@ pub fn remove(options: &RemoveOptions<'_>) -> CargoResult<()> {
     let manifest_path = options.spec.manifest_path().to_path_buf();
     let mut manifest = LocalManifest::try_new(&manifest_path)?;
 
+    let mut workspace_manifest = if workspace.is_virtual() {
+        let data = cargo_util::paths::read(workspace.root_manifest())?;
+        let manifest: toml_edit::Document = data.parse()?;
```

Comment:
There a reason you are loading a `toml_edit::Document` directly rather than a `LocalManifest`?

Reply:
See above comment:

> A lot of this work mimics some of the work done in `cargo::util::toml_mut::manifest`, which explicitly does not support virtual manifests. Alternatively, we could expand `toml_mut` in some way to support this, but given that the surface area of this behavior is relatively small, it may not be worth it to do that.